src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Application/Commands/AddCohortCommandHandlerTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.CommitmentsV2.Application.Commands.AddCohort;
+using SFA.DAS.CommitmentsV2.Application.Commands.CreateAccount;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Exceptions;
+using SFA.DAS.CommitmentsV2.Models;
+using SFA.DAS.ProviderCommitments.HashingTemp;
+using SFA.DAS.Testing;
+
+namespace SFA.DAS.CommitmentsV2.UnitTests.Application.Commands
+{
+    [TestFixture]
+    [Parallelizable]
+    public class AddCohortCommandHandlerTests : FluentTest<AddCohortCommandHandlerTestFixture>
+    {
+        [Test]
+        public async Task Handle_WhenHandlingValidCommand_ThenShouldCreateAccountAndLogSuccess()
+        {
+            const string expectedHash = "ABC123";
+
+            var fixtures = new AddCohortCommandHandlerTestFixture()
+                                .WithGeneratedHash(expectedHash)
+                                .WithProvider(1)
+                                .WithAccountLegalEntity(2, 3);
+
+            var response = await fixtures.Handle(3, 1);
+
+            Assert.IsNotNull(response);
+            Assert.AreNotEqual(0, response.Id);
+            Assert.AreNotEqual(0, response.DraftApprenticeshipId);
+            Assert.AreEqual(expectedHash, response.Reference);
+             
+            Assert.IsTrue(fixtures.Logger.HasInfo);
+            Assert.IsFalse(fixtures.Logger.HasErrors);
+        }
```

Comment:
I think this is the testing-the-implementation vs testing-the-behaviour argument. The code under test is required to provide a new instance with an ID - so that's what the test checks. Checking that save is called is checking the implementation of how it achieves this goal. As it happens the code wouldn't get an Id if the save was called successfully.